File: resources/galaxia_resources/code/java/buildings/Mine.java
package buildings;

public class Mine extends Structure {
    private int ProductionRate = 1;
    private int BaseRate = 100;

    public Mine() {
        super(500);
    }

    public double Proc() {
        return super.Proc(this.BaseRate + (this.Level - 1) * this.Multiplier * this.ProductionRate);
    }

}

File: models/buildings/mine.py
from structure import Structure

class Mine(Structure):

    production_rate: int = 1
    base_rate: int = 100

    def __init__(self):
        super(500)

    def proc(self) -> float:
        return super().proc(self.base_rate + (self.level - 1) * self.multiplier * self.production_rate)